client/cpp/kserve-api/samples/grpc_async_infer_resnet.cpp

```diff
@@ -0,0 +1,204 @@
+    std::vector<int> labels;
+    std::ifstream images(args["images_list"].as<std::string>());
+    while (images >> img >> label) {
+        imgs.push_back(img);
+        labels.push_back(label);
+    }
+
+    std::vector<int64_t> shape{1};
+
+    // Initialize the inputs with the data.
+    tc::InferInput* input;
+
+    FAIL_IF_ERR(
+        tc::InferInput::Create(&input, input_name, shape, "BYTES"),
+        "unable to get input");
+    std::shared_ptr<tc::InferInput> input_ptr;
+    input_ptr.reset(input);
+
+    tc::InferOptions options(model_name);
+    if (args.count("model_version"))
+        options.model_version_ = args["model_version"].as<std::string>();
+    try {
+        options.client_timeout_ = args["timeout"].as<int>();
+    } catch (cxxopts::argument_incorrect_type e) {
+        std::cout << "The provided argument is of a wrong type" << std::endl;
+        return 1;
+    }
+    std::vector<tc::InferInput*> inputs = {input_ptr.get()};
+
+    std::vector<std::string> classes;
+    std::ifstream lb_f(args["labels_list"].as<std::string>());
+    std::string tmp;
+    while (std::getline(lb_f, tmp)) {
+        classes.push_back(tmp);
+    }
+
+    std::vector<std::vector<uint8_t>> input_data; 
+    input_data.reserve(imgs.size());
+    for (int i = 0; i < imgs.size(); i++) {
+        input_data.push_back(Load(imgs[i]));
+    }
+
+    int acc = 0;
+    std::mutex mtx;
+    std::condition_variable cv;
+    int c = 0;
+    for (int i = 0; i < imgs.size(); i++) {
+        FAIL_IF_ERR(
+            input_ptr->AppendRaw(input_data[i]),
+            "unable to set data for input");
+        client->AsyncInfer(
+            [&, i](tc::InferResult* result){
+                {
+                    std::shared_ptr<tc::InferResult> result_ptr;
+                    result_ptr.reset(result);
+                    std::lock_guard<std::mutex> lk(mtx);
+                    c++;
+                    // Get pointers to the result returned...
+                    float* output_data;
+                    size_t output_byte_size;
+                    FAIL_IF_ERR(
+                        result_ptr->RawData(
+                            output_name, (const uint8_t**)&output_data, &output_byte_size),
+                        "unable to get result data for output");
+
+                    int lb = std::distance(output_data, std::max_element(output_data, output_data + 1000));
+                    std::cout << imgs[i] << " classified as "
+                            << lb << " " << classes[lb] << " ";
+                    if (lb != labels[i]) {
+                        std::cout << "should be " << labels[i] << " " << classes[labels[i]];
+                    } else {
+                        acc++;
+                    }
+                    std::cout << std::endl;
+                }
+                cv.notify_all();
+            }, options, inputs);
+        input->Reset();
+    }
+    {
+        std::unique_lock<std::mutex> lk(mtx);
```

Comment:
Fix indentation, possibly use make style? cppstyle and cpplint.
To me, it looks like L180-L187 is in one indent